bserver/bendoapi.go

```diff
@@ -41,6 +44,53 @@ func (ia *itemAttributes) GetItemInfo() (*jason.Object, error) {
 	return v, err
 }
 
+func (ia *itemAttributes) downLoad(fileName string, pathPrefix string) error {
+	var httpPath = "http://" + ia.bendoServer + "/item/" + ia.item + "/" + fileName
+
+	r, err := http.Get(httpPath)
+	if err != nil {
+		return err
+	}
+	if r.StatusCode != 200 {
+		r.Body.Close()
+		switch r.StatusCode {
+		case 404:
+			return ErrNotFound
+		case 401:
+			return ErrNotAuthorized
+		default:
+			return fmt.Errorf("Received status %d from Bendo", r.StatusCode)
+		}
+	}
+
+	// How do we handle large downloads?
+
+	targetFile := path.Join(pathPrefix, fileName)
+
+	targetDir, _ := path.Split(targetFile)
+
+	err = os.MkdirAll(targetDir, 0755)
+
+	if err != nil {
+		fmt.Printf("Error: could not create directory %s\n%s\n", err.Error())
+		return err
+	}
+
+	filePtr, err := os.Create(targetFile)
+
+	if err != nil {
+		fmt.Printf("Error: could not create file %s\n%s\n", err.Error())
+		return err
+	}
+
+	defer r.Body.Close()
```

Comment:
the `defer r.Body.Close()` should be moved up (to line 65) since there are returns in the middle, so there are situations where the body is not closed.